File: 1200/1272C/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int n, k;
    cin >> n >> k;

    string s;
    cin >> s;

    vector<int> allowed(26, 0);
    for (int i = 0; i < k; i++) 
    {
        char c;
        cin >> c;
        allowed[c - 'a'] = 1;
    }

    long long sum = 0;
    long long ans = 0;

    for (int i = 0; i < n; i++) 
    {
        if (allowed[s[i] - 'a'])
            sum++;
        else 
        {
            ans += (sum * (sum + 1)) / 2;
            sum = 0;
        }
    }

    ans += (sum * (sum + 1)) / 2;
    cout << ans;
}

File: 1200/1272C/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        
        String s = sc.next();
        
        int ch[] = new int[26];
        for(int i=0;i<k;i++)
            ch[sc.next().charAt(0)-'a']++;
        
        long sum = 0;
        long ans = 0;
        
        for(int i=0;i<n;i++)
        {
            if(ch[s.charAt(i)-'a'] > 0)
                sum++;
            else
            {
                ans += (sum*(sum+1))/2;
                sum = 0;
            }
        }
        
        System.out.println(ans + (sum*(sum+1))/2);
    }
}